### PR TITLE
youtube-dl: version bump.

### PR DESCRIPTION
**Pull request by netpositive36**

### CODE REVIEW

File: net-misc/youtube_dl/youtube_dl-2017.11.15.recipe [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+BUILD_PREREQUIRES="
+	cmd:zip
+	"
+
+BUILD()
+{
+	zip -q youtube-dl youtube_dl/*.py youtube_dl/*/*.py
+	zip -q -j youtube-dl youtube_dl/__main__.py
+	echo '#!/bin/python' > youtube-dl
+	cat youtube-dl.zip >> youtube-dl
```
Comment (extrowerk) [inline net-misc/youtube_dl/youtube_dl-2017.11.15.recipe]: Is theese things actually required?

Comment (extrowerk) [reply net-misc/youtube_dl/youtube_dl-2017.11.15.recipe]: It comes with a just absolutely normal setup.py, why is this hickety-hack required?
If it is about the .pyc files, one can delete them as last step in the INSTAL()

File: net-misc/youtube_dl/youtube_dl-2017.11.15.recipe [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+INSTALL()
+{
+	mkdir -p $binDir
+	cp -a youtube-dl $binDir
+}
+
+TEST()
+{
+	youtube-dl --version
```
Comment (extrowerk) [inline net-misc/youtube_dl/youtube_dl-2017.11.15.recipe]: Nice, but youtube-dl provides plenty testcases in the test folder.
I fear, this TEST() section doesn't cover much functionality.

**Comment by korli** [comment]
@extrowerk to be fair, this recipe exists already https://github.com/haikuports/haikuports/blob/master/net-misc/youtube_dl/youtube_dl-2017.10.29.recipe
I think I tried the setup.py one time, it didn't really work, can't remember, or lazyness :)

**Comment by extrowerk** [comment]
@korli I know and i checked it as i made my review. I see 2 different ways to clean the recipe up:
- Keep the current install mechanic, but clean up the REQs (easier)
- Explore the normal setup.py way, rework the complete recipe according existing python based recipes (could be problematic)

But currently it is referencing way too much unrequired deps for my taste.

**Comment by netpositive36** [comment]
@korli @extrowerk Thanks for the review, I'm going to clean up the recipe. What do you think about removing the old recipes? I see no reason to keep them for this app.

**Comment by extrowerk** [comment]
It seems ubertuber using youtube-dl directly, so you should check ubertuber first. If it still works with the current youtube-dl you can delete the old recipe, as no other program using it.

File: net-misc/youtube_dl/youtube_dl-2017.11.15.recipe [hunk 3]
```diff
@@ -26,20 +25,16 @@ REQUIRES="
...
 	"
 
 BUILD()
 {
-	zip -q youtube-dl youtube_dl/*.py youtube_dl/*/*.py
-	zip -q -j youtube-dl youtube_dl/__main__.py
-	echo '#!/bin/python' > youtube-dl
-	cat youtube-dl.zip >> youtube-dl
-	rm youtube-dl.zip
+	sed 's/\/usr//' youtube-dl > youtube-dl.tmp
```
Comment (extrowerk) [inline net-misc/youtube_dl/youtube_dl-2017.11.15.recipe]: Why not 'in-place'?

File: net-misc/youtube_dl/youtube_dl-2017.11.15.recipe [hunk 4]
```diff
@@ -26,20 +25,16 @@ REQUIRES="
 	"
 
 BUILD_REQUIRES="
-	haiku_devel
-	cmd:python
 	"
+
 BUILD_PREREQUIRES="
 	cmd:zip
```
Comment (extrowerk) [inline net-misc/youtube_dl/youtube_dl-2017.11.15.recipe]: Not required anymore.

**Pull request merged**